### DISCUSSION TITLE
The typography plugin is overriding my custom plugin

### DISCUSSION THREAD

**Discussion by JonathanMMunoz** [tailwindlabs/tailwindcss#4446] [Help] [answered]
I am currently using the typography plugin, but it seems to be overriding the CSS-in-JS plugin I have for my buttons:

```
const plugin = require("tailwindcss/plugin");
const colors = require("tailwindcss/colors");

module.exports = {
  theme: {
    container: {
      center: true,
    },
    extend: {
      fontFamily: {
        sans: ["Roboto", "system-ui"],
      },
      typography: {
        sm: {
          css: {
            h1: {
              fontWeight: "700",
            },
          },
        },
        DEFAULT: {
          css: {
            h1: {
              fontSize: "3.75rem",
              lineHeight: "1.5",
            },
          },
        },
      },
    },
  },
  variants: {
    extend: {
      borderColor: ["focus-visible"],
      opacity: ["disabled"],
    },
  },
  plugins: [
    require("@tailwindcss/typography"),
    require("@tailwindcss/forms"),
    require("@tailwindcss/aspect-ratio"),
    plugin(function ({ addComponents, theme }) {
      const buttons = {
        ".btn": {
          backgroundColor: theme("colors.indigo.500"),
          padding: `${theme("spacing.2")} ${theme("spacing.4")}`,
          color: theme("colors.white"),
          fontWeight: theme("fontWeight.600"),
          "&:hover": {
            backgroundColor: theme("colors.white"),
            color: theme("colors.indigo.500"),
          },          
        },
        ".btn-muted": {
          backgroundColor: theme("colors.black"),
          "&:hover": {
            backgroundColor: theme("colors.indigo.600"),
          },
        },
      };
      addComponents(buttons);
    }),
  ],
};
```

Is t here a way to have my plugin to be prioritized over the changes the typography plugin brings to my build?

**Comment by simonswiss** [comment, marked as answer]
Hey!

I guess that happens if you use an `<a>` tag with a class of `btn`, since the Typography plugin applies default styles to the `a` HTML element.

Styles applied from the Typography plugin will override a single class component, since they are applied through a **nested selector**, using the `prose` parent class. The `.prose a` selector will have more specificity than the `a.btn` selector, and therefore win.

The Typography plugin is typically meant to be used in situations where you _cannot_ really control the HTML output, and expect some raw HTML markup. It's not always working well with elements where you would apply CSS classes. If you want default typographic styles applied to your entire project (as opposed to being scoped inside a `prose` class), you could take inspiration from the [styles generated by that plugin](https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/styles.js), and define those in your `@base` layer instead, [using the `addBase` function](https://tailwindcss.com/docs/adding-base-styles#using-a-plugin) from the plugin API.

If you really want/need to use the Typography plugin, I can think of a few options here (neither being ideal):

1) Try to have your styled link (`.btn`) sit outside of the parent element where your `prose` class is applied, if possible.

2) If that's not possible, identify what styles are clashing (styles applied to the `a` element is an example), and define those styles in the Typography plugin in your config with the `.btn` class appended to it:

```diff
DEFAULT: {
  css: {
    h1: {
      fontSize: '3.75rem',
      lineHeight: '1.5',
    },
+  'a.btn': { // add your styles here },
  },
},
```

3) Add more specificity/overrides to your button plugin by adding the prose selector as well as just the `.btn` class:

```diff
plugin(function ({ addComponents, theme }) {
  const buttons = {
+    '.btn, .prose .btn': {
```

... or by adding `!important` to your button color (and other properties needed).

Like I said, neither of 2) and 3) are ideal, the _real_ way to fix this would be to apply your typographic styles to the `@base` layer instead of through the `prose` scoped class, if you intend to apply your own classes to that markup inside the `prose` parent.

One more option: if you're using the `JIT` mode, you can use the `!` important prefix to any class if you need to override specific properties with a single utility. ❗️

Hope that helps!

**Reply by MartinMa** [reply]
Thanks for your explanation and your solution approaches.

I'm using option 3) to override certain `a` styles. At everyone: Make sure to use `addComponents` instead of `addUtilities`. That way, you should still be able to override your component styles with utility classes later. Even if you don't plan to override the styles, it makes more sense to keep the styles in the components layer.
